packages/catalog-process/src/services/catalogService.ts

```diff
@@ -625,212 +548,211 @@ export function catalogServiceBuilder(
+          producerId: authData.organizationId,
+        });
+      if (eserviceWithSameName) {
+        throw eServiceDuplicate(apiEServicesSeed.name);
+      }
 
-  return toCreateEventEServiceAdded(newEService);
-}
+      const newEService: EService = {
+        id: generateId(),
+        producerId: authData.organizationId,
+        name: apiEServicesSeed.name,
+        description: apiEServicesSeed.description,
+        technology: apiTechnologyToTechnology(apiEServicesSeed.technology),
+        mode: apiEServiceModeToEServiceMode(apiEServicesSeed.mode),
+        attributes: undefined,
+        descriptors: [],
+        createdAt: new Date(),
+        riskAnalysis: [],
+      };
 
-export async function updateEserviceLogic({
-  eService,
-  eserviceId,
-  authData,
-  eServiceSeed,
-  getEServiceByNameAndProducerId,
-  deleteFile,
-}: {
-  eService: WithMetadata<EService> | undefined;
-  eserviceId: EServiceId;
-  authData: AuthData;
-  eServiceSeed: ApiEServiceSeed;
-  getEServiceByNameAndProducerId: ({
-    name,
-    producerId,
-  }: {
-    name: string;
-    producerId: TenantId;
-  }) => Promise<WithMetadata<EService> | undefined>;
-  deleteFile: (container: string, path: string) => Promise<void>;
-}): Promise<CreateEvent<EServiceEvent>> {
-  assertEServiceExist(eserviceId, eService);
-  assertRequesterAllowed(eService.data.producerId, authData.organizationId);
+      const event = toCreateEventEServiceAdded(newEService);
+      await repository.createEvent(event);
 
-  if (
-    !(
-      eService.data.descriptors.length === 0 ||
-      (eService.data.descriptors.length === 1 &&
-        eService.data.descriptors[0].state === descriptorState.draft)
-    )
-  ) {
-    throw eServiceCannotBeUpdated(eserviceId);
-  }
+      return newEService;
+    },
 
-  if (eServiceSeed.name !== eService.data.name) {
-    const eServiceWithSameName = await getEServiceByNameAndProducerId({
-      name: eServiceSeed.name,
-      producerId: authData.organizationId,
-    });
-    if (eServiceWithSameName !== undefined) {
-      throw eServiceDuplicate(eServiceSeed.name);
-    }
-  }
+    async updateEService(
+      eserviceId: EServiceId,
+      eServiceSeed: ApiEServiceSeed,
+      authData: AuthData
+    ): Promise<EService> {
+      logger.info(`Updating EService ${eserviceId}`);
 
-  const updatedTechnology = apiTechnologyToTechnology(eServiceSeed.technology);
-  if (eService.data.descriptors.length === 1) {
-    const draftDescriptor = eService.data.descriptors[0];
-    if (
-      updatedTechnology !== eService.data.technology &&
-      draftDescriptor.interface !== undefined
-    ) {
-      await deleteFile(config.s3Bucket, draftDescriptor.interface.path).catch(
-        (error) => {
-          logger.error(
-            `Error deleting interface for descriptor ${draftDescriptor.id} : ${error}`
-          );
-          throw error;
+      const eService = await retrieveEService(eserviceId, readModelService);
```

Comment:
The function `retrieveEService` includes the throwing of `eServiceNotFound`, so the call to `assertEServiceExist` is no longer needed.